include/barrier.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public interface for barriers.
+ */
+#ifndef ZEPHYR_INCLUDE_BARRIER_H_
+#define ZEPHYR_INCLUDE_BARRIER_H_
+
+#ifndef _ASMLANGUAGE
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup barrier_apis Barriers API
+ * @ingroup kernel_apis
+ * @{
+ */
+
+#if !defined(nop) || defined(__DOXYGEN__)
+/**
+ * @brief nop
+ *
+ * NOP instruction.
+ */
+#define nop()		__asm__ volatile ("nop")
```

Comment:
What's nop doing here?  Also don't we already have an arch_nop()?   And while nothing comes to mind, I'm sure there is, somewhere, an architecture where a "nop" symbol doesn't emit an instruction that does a nop.

Reply:
ACK